frontend/src/common/ReviewWithComment.tsx

```diff
@@ -90,66 +90,68 @@ export default function ReviewWithComment({
 
   return (
     <>
-      {isModelRolesLoading && <Loading />}
-      <Dialog fullWidth open={open} onClose={onClose}>
-        <DialogTitle>{title}</DialogTitle>
-        <DialogContent>
-          <Stack spacing={2}>
-            <Autocomplete
-              sx={{ pt: 1 }}
-              open={selectOpen}
-              onOpen={() => {
-                setSelectOpen(true)
-              }}
-              onClose={() => {
-                setSelectOpen(false)
-              }}
-              // we might get a string or an object back
-              isOptionEqualToValue={(option: ReviewRequestInterface, value: ReviewRequestInterface) =>
-                option.role === value.role
-              }
-              onChange={onChange}
-              value={reviewRequest}
-              disabled={reviews.length === 1}
-              getOptionLabel={(option) => getRoleDisplay(option.role, modelRoles)}
-              options={reviews}
-              renderInput={(params) => <TextField {...params} label='Select your role' size='small' />}
-            />
-            {description && <DialogContentText>{description}</DialogContentText>}
-            <TextField
-              size='small'
-              minRows={4}
-              maxRows={8}
-              multiline
-              placeholder='Leave a comment'
-              data-test='review-with-comment-input'
-              value={reviewComment}
-              onChange={(e) => setReviewComment(e.target.value)}
-              error={showError}
-              helperText={showError ? 'You must submit a comment when requesting changes.' : ''}
-            />
-            <Stack
-              spacing={2}
-              direction={{ sm: 'row', xs: 'column' }}
-              justifyContent='space-between'
-              alignItems='center'
-            >
-              <Button onClick={onClose}>Cancel</Button>
-              <Stack spacing={2} direction={{ sm: 'row', xs: 'column' }}>
-                <Button variant='outlined' onClick={() => submitForm(ResponseTypes.RequestChanges)}>
-                  Request Changes
-                </Button>
-                <Button variant='contained' onClick={() => submitForm(ResponseTypes.Approve)}>
-                  Approve
-                </Button>
+      {!modelRoles.length && isModelRolesLoading && <Loading />}
```

Comment:
I don't think we need `!modelRoles.length` in this check. The loading spinner is there purely to inform the user if data if being fetched. The `models.length` check below should be enough to stop the crashing.